src/server/game/Spells/Spell.cpp

```diff
@@ -5498,7 +5498,7 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
             case SPELL_AURA_MOUNTED:
             {
-                if (m_caster->IsInWater())
+                if (m_caster->IsInWater() && m_spellInfo->HasAura(SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED))
```

Comment:
I would suggest just moving this statement if it's correct to the SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED case ~30 lines below.

On an unrelated note: I could swear I remember it only letting me mount while I was at the top of the water line on retail during WoTLK, however I cannot find any proof of this. Older patch perhaps or am I just not remembering things correctly?